src/tokenProvider/kmsTokenProvider.ts

```diff
@@ -0,0 +1,60 @@
+import { KMS } from 'aws-sdk';
+import TokenProvider, {
+  Auth0Secret,
+  TokenConfiguration,
+  TokenProviderOptions,
+} from './tokenProvider';
+
+export default class KmsTokenProvider extends TokenProvider {
+  private kmsClient: KMS;
+
+  private kmsConfiguration: KmsTokenConfiguration;
+
+  constructor(options: KmsTokenProviderOptions) {
+    if (!options.kmsClient) {
+      throw new Error('Configuration error: missing required property "kmsClient"');
+    }
+    if (!options.tokenConfiguration?.clientId) {
+      throw new Error('Configuration error: missing required property "clientId"');
+    }
+    if (!options.tokenConfiguration?.encryptedClientSecret) {
+      throw new Error('Configuration error: missing required property "encryptedClientSecret"');
+    }
```

Comment:
Actually, I think we can drop all the validations because TypeScript will already take care of them? It might be that they got here by copy-paste from a previous JS project where this was implemented.